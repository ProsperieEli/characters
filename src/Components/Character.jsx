import React from "react";

export default function Character({ head, middle, bottom }) {
  return (
    <div>
      <section style={{ display: "flex", flexDirection: "column" }}>
        <article
          className="head"
          aria-label="head"
          style={{
            backgroundImage: `url(./${head}-head.png)`,
            height: 150,
            width: 150,
            backgroundSize: "cover",
            backgroundRepeat: "no-repeat",
          }}
        ></article>

        <article
          className="middle"
          style={{
            backgroundImage: `url(./${middle}-middle.png)`,
            height: 150,
            width: 150,
            backgroundSize: "cover",
            backgroundRepeat: "no-repeat",
          }}
        ></article>
        <article
          className="bottom"
          style={{
            backgroundImage: `url(./${bottom}-bottom.png)`,
            height: 150,
            width: 150,
            backgroundSize: "cover",
            backgroundRepeat: "no-repeat",
          }}
        ></article>
      </section>
    </div>
  );
}
