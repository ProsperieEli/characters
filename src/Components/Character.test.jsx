import { screen, render } from "@testing-library/react";
import Character from "./Character";

it("Should display the Character component", () => {
  const { container } = render(
    <Character head={"dog"} middle={"fancy"} bottom={"blue"} />
  );

  const head = screen.getByLabelText(/head/);
  const middle = screen.getByLabelText(/middle/);
  const bottom = screen.getByLabelText(/bottom/);

  expect(head).toHaveStyle(`background-image: url(./dog-head.png)`);
  expect(middle).toHaveStyle(`background-image: url(./fancy-middle.png)`);
  expect(bottom).toHaveStyle(`background-image: url(./blue-bottom.png)`);
  expect(container).toMatchSnapshot();
});
