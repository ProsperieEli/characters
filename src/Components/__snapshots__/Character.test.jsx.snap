// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the Character component 1`] = `
<div>
  <div>
    <section
      style="display: flex; flex-direction: column;"
    >
      <article
        aria-label="head"
        class="head"
        style="background-image: url(./dog-head.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
      />
      <article
        aria-label="middle"
        class="middle"
        style="background-image: url(./fancy-middle.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
      />
      <article
        aria-label="bottom"
        class="bottom"
        style="background-image: url(./blue-bottom.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
      />
    </section>
  </div>
</div>
`;
